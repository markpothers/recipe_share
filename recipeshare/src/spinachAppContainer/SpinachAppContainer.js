import React from 'react'
import { ScrollView, SafeAreaView, Image, KeyboardAvoidingView, View, ActivityIndicator, Platform } from 'react-native'
import { centralStyles } from '../centralStyleSheet'


export default class SpinachAppContainer extends React.Component {
    static navigationOptions = {

    }

    state = {
        scrollingEnabled: true
    }

    render() {
        if (this.props.scrollingEnabled){
            return (
                <SafeAreaView style={centralStyles.fullPageSafeAreaView}>
                    <KeyboardAvoidingView style={centralStyles.fullPageKeyboardAvoidingView} behavior={(Platform.OS === "ios" ? "padding" : "")}>
                        <Image
                            source={require('../dataComponents/spinach.jpg')}
                            style={centralStyles.spinachFullBackground}
                        />
                        {this.props.awaitingServer && <View style={centralStyles.activityIndicatorContainer}><ActivityIndicator style={(Platform.OS === 'ios' ? centralStyles.activityIndicator : {})} size="large" color="#104e01" /></View>}
                        <ScrollView style={centralStyles.fullPageScrollView}
                            nestedScrollEnabled={true}
                            scrollEnabled={this.state.scrollingEnabled}
                            keyboardShouldPersistTaps={'always'}
                            ref={(c) => { this.scrollView = c; }}
                        >
                            {this.props.children}
                        </ScrollView>
                    </KeyboardAvoidingView>
                </SafeAreaView>
            )
        }
        else {
            return (
                <SafeAreaView style={centralStyles.fullPageSafeAreaView}>
                    <Image
                        source={require('../dataComponents/spinach.jpg')}
                        style={centralStyles.spinachFullBackground}
                    />
                    {this.props.awaitingServer && <View style={centralStyles.activityIndicatorContainer}><ActivityIndicator style={(Platform.OS === 'ios' ? centralStyles.activityIndicator : {})} size="large" color="#104e01" /></View>}
                    {this.props.children}
                </SafeAreaView>
            )
        }
    }
}