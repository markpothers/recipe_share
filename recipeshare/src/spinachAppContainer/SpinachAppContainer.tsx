import { Image, KeyboardAvoidingView, Platform, SafeAreaView, ScrollView } from "react-native"
import React, { useState } from "react"

import StyledActivityIndicator from "../customComponents/styledActivityIndicator/styledActivityIndicator"
import { centralStyles } from "../centralStyleSheet" //eslint-disable-line no-unused-vars

type OwnProps = {
	scrollingEnabled: boolean,
	awaitingServer: boolean,
	children: React.ReactElement[]
}

const SpinachAppContainer = ({scrollingEnabled, awaitingServer, children}: OwnProps) => {
// export default class SpinachAppContainer extends React.Component {
	const [ stateScrollingEnabled ] = useState(true)

		if (scrollingEnabled) {
			return (
				<SafeAreaView style={centralStyles.fullPageSafeAreaView}>
					<KeyboardAvoidingView
						style={centralStyles.fullPageKeyboardAvoidingView}
						// @ts-ignore
						behavior={(Platform.OS === "ios" ? "padding" : "")}
						keyboardVerticalOffset={Platform.OS === "ios" ? 20 : 0}
					>
						<Image
							source={require("../dataComponents/spinach.jpg")}
							style={centralStyles.spinachFullBackground}
							resizeMode={"cover"}
						/>
						{awaitingServer && <StyledActivityIndicator/>}
						<ScrollView style={centralStyles.fullPageScrollView}
							nestedScrollEnabled={true}
							scrollEnabled={stateScrollingEnabled}
							keyboardShouldPersistTaps={"always"}
							// ref={(c) => { this.scrollView = c }}
						>
							{children}
						</ScrollView>
					</KeyboardAvoidingView>
				</SafeAreaView>
			)
		}
		else {
			return (
				<SafeAreaView style={centralStyles.fullPageSafeAreaView}>
					<Image
						source={require("../dataComponents/spinach.jpg")}
						style={centralStyles.spinachFullBackground}
						resizeMode={"cover"}
					/>
					{awaitingServer && <StyledActivityIndicator/>}
					{children}
				</SafeAreaView>
			)
		}
	}

export default SpinachAppContainer
