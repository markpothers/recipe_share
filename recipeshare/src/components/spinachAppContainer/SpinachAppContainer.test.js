import { ActivityIndicator, Platform, ScrollView } from "react-native";
import renderer, { act } from "react-test-renderer";

import React from "react";
import SpinachAppContainer from "./SpinachAppContainer";

describe("SpinachAppContainer", () => {
	let component;

	beforeAll(() => {
		// console.log('runs at the beginning of everything')
	});

	beforeEach(() => {
		// console.log('runs before every test')
	});

	afterEach(() => {
		// console.log('runs after each test')
	});

	afterAll(() => {
		// console.log('runs after all tests have completed')
	});

	test("renders with scrollView on IOS", () => {
		act(() => {
			component = renderer.create(<SpinachAppContainer scrollingEnabled={true} awaitingServer={false} />);
		});
		const image = component.toJSON();
		expect(image).toMatchSnapshot();
		let root = component.root;
		const scrollView = root.findAllByType(ScrollView);
		expect(scrollView.length).toEqual(1);
	});

	test("renders without scrollView on IOS", () => {
		act(() => {
			component = renderer.create(<SpinachAppContainer scrollingEnabled={false} awaitingServer={false} />);
		});
		const image = component.toJSON();
		expect(image).toMatchSnapshot();
		let root = component.root;
		const scrollView = root.findAllByType(ScrollView);
		expect(scrollView.length).toEqual(0);
	});

	test("renders with scrollView and activityIndicator on IOS", () => {
		act(() => {
			component = renderer.create(<SpinachAppContainer scrollingEnabled={true} awaitingServer={true} />);
		});
		const image = component.toJSON();
		expect(image).toMatchSnapshot();
		let root = component.root;
		const scrollView = root.findAllByType(ScrollView);
		expect(scrollView.length).toEqual(1);
		const indicator = root.findAllByType(ActivityIndicator);
		expect(indicator.length).toEqual(1);
	});

	test("renders without scrollView and with ActivityIndicator on IOS", () => {
		act(() => {
			component = renderer.create(<SpinachAppContainer scrollingEnabled={false} awaitingServer={true} />);
		});
		const image = component.toJSON();
		expect(image).toMatchSnapshot();
		let root = component.root;
		const scrollView = root.findAllByType(ScrollView);
		expect(scrollView.length).toEqual(0);
		const indicator = root.findAllByType(ActivityIndicator);
		expect(indicator.length).toEqual(1);
	});

	test("renders with scrollView on Android", () => {
		Platform.OS = "android";
		act(() => {
			component = renderer.create(<SpinachAppContainer scrollingEnabled={true} awaitingServer={false} />);
		});
		const image = component.toJSON();
		expect(image).toMatchSnapshot();
		let root = component.root;
		const scrollView = root.findAllByType(ScrollView);
		expect(scrollView.length).toEqual(1);
	});

	test("renders without scrollView on Android", () => {
		Platform.OS = "android";
		act(() => {
			component = renderer.create(<SpinachAppContainer scrollingEnabled={false} awaitingServer={false} />);
		});
		const image = component.toJSON();
		expect(image).toMatchSnapshot();
		let root = component.root;
		const scrollView = root.findAllByType(ScrollView);
		expect(scrollView.length).toEqual(0);
	});

	test("renders with scrollView and activityIndicator on Android", () => {
		Platform.OS = "android";
		act(() => {
			component = renderer.create(<SpinachAppContainer scrollingEnabled={true} awaitingServer={true} />);
		});
		const image = component.toJSON();
		expect(image).toMatchSnapshot();
		let root = component.root;
		const scrollView = root.findAllByType(ScrollView);
		expect(scrollView.length).toEqual(1);
		const indicator = root.findAllByType(ActivityIndicator);
		expect(indicator.length).toEqual(1);
	});

	test("renders without scrollView and with ActivityIndicator on Android", () => {
		Platform.OS = "android";
		act(() => {
			component = renderer.create(<SpinachAppContainer scrollingEnabled={false} awaitingServer={true} />);
		});
		const image = component.toJSON();
		expect(image).toMatchSnapshot();
		let root = component.root;
		const scrollView = root.findAllByType(ScrollView);
		expect(scrollView.length).toEqual(0);
		const indicator = root.findAllByType(ActivityIndicator);
		expect(indicator.length).toEqual(1);
	});
});
