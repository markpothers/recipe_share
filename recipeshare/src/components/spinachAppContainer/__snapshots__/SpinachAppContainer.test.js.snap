// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpinachAppContainer renders with scrollView and activityIndicator on Android 1`] = `
<RCTSafeAreaView
  style={
    {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "red",
        "borderWidth": 0,
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        {
          "borderColor": "black",
          "borderWidth": 0,
          "height": 1334,
          "position": "absolute",
          "width": 750,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "left": "50%",
          "marginLeft": -25,
          "position": "absolute",
          "top": "10%",
          "width": 50,
          "zIndex": 5,
        }
      }
    >
      <ActivityIndicator
        color="#104e01"
        size="large"
        style={{}}
        testID="activityIndicator"
      />
    </View>
    <RCTScrollView
      keyboardShouldPersistTaps="always"
      nestedScrollEnabled={true}
      scrollEnabled={true}
      style={
        {
          "borderColor": "yellow",
          "borderWidth": 0,
          "flex": 1,
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`SpinachAppContainer renders with scrollView and activityIndicator on IOS 1`] = `
<RCTSafeAreaView
  style={
    {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "red",
          "borderWidth": 0,
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        {
          "borderColor": "black",
          "borderWidth": 0,
          "height": 1334,
          "position": "absolute",
          "width": 750,
        }
      }
    />
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#fff59b",
          "borderColor": "#104e01",
          "borderRadius": 11.25,
          "borderWidth": 1,
          "height": 50,
          "justifyContent": "center",
          "left": "50%",
          "marginLeft": -25,
          "position": "absolute",
          "top": "10%",
          "width": 50,
          "zIndex": 5,
        }
      }
    >
      <ActivityIndicator
        color="#104e01"
        size="large"
        style={
          {
            "left": 1.25,
            "top": 1.7,
          }
        }
        testID="activityIndicator"
      />
    </View>
    <RCTScrollView
      keyboardShouldPersistTaps="always"
      nestedScrollEnabled={true}
      scrollEnabled={true}
      style={
        {
          "borderColor": "yellow",
          "borderWidth": 0,
          "flex": 1,
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`SpinachAppContainer renders with scrollView on Android 1`] = `
<RCTSafeAreaView
  style={
    {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      {
        "alignItems": "center",
        "borderColor": "red",
        "borderWidth": 0,
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        {
          "borderColor": "black",
          "borderWidth": 0,
          "height": 1334,
          "position": "absolute",
          "width": 750,
        }
      }
    />
    <RCTScrollView
      keyboardShouldPersistTaps="always"
      nestedScrollEnabled={true}
      scrollEnabled={true}
      style={
        {
          "borderColor": "yellow",
          "borderWidth": 0,
          "flex": 1,
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`SpinachAppContainer renders with scrollView on IOS 1`] = `
<RCTSafeAreaView
  style={
    {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      [
        {
          "alignItems": "center",
          "borderColor": "red",
          "borderWidth": 0,
          "flex": 1,
        },
        {
          "paddingBottom": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        {
          "borderColor": "black",
          "borderWidth": 0,
          "height": 1334,
          "position": "absolute",
          "width": 750,
        }
      }
    />
    <RCTScrollView
      keyboardShouldPersistTaps="always"
      nestedScrollEnabled={true}
      scrollEnabled={true}
      style={
        {
          "borderColor": "yellow",
          "borderWidth": 0,
          "flex": 1,
        }
      }
    >
      <View />
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;

exports[`SpinachAppContainer renders without scrollView and with ActivityIndicator on Android 1`] = `
<RCTSafeAreaView
  style={
    {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      {
        "borderColor": "black",
        "borderWidth": 0,
        "height": 1334,
        "position": "absolute",
        "width": 750,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff59b",
        "borderColor": "#104e01",
        "borderRadius": 11.25,
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "left": "50%",
        "marginLeft": -25,
        "position": "absolute",
        "top": "10%",
        "width": 50,
        "zIndex": 5,
      }
    }
  >
    <ActivityIndicator
      color="#104e01"
      size="large"
      style={{}}
      testID="activityIndicator"
    />
  </View>
</RCTSafeAreaView>
`;

exports[`SpinachAppContainer renders without scrollView and with ActivityIndicator on IOS 1`] = `
<RCTSafeAreaView
  style={
    {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      {
        "borderColor": "black",
        "borderWidth": 0,
        "height": 1334,
        "position": "absolute",
        "width": 750,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#fff59b",
        "borderColor": "#104e01",
        "borderRadius": 11.25,
        "borderWidth": 1,
        "height": 50,
        "justifyContent": "center",
        "left": "50%",
        "marginLeft": -25,
        "position": "absolute",
        "top": "10%",
        "width": 50,
        "zIndex": 5,
      }
    }
  >
    <ActivityIndicator
      color="#104e01"
      size="large"
      style={
        {
          "left": 1.25,
          "top": 1.7,
        }
      }
      testID="activityIndicator"
    />
  </View>
</RCTSafeAreaView>
`;

exports[`SpinachAppContainer renders without scrollView on Android 1`] = `
<RCTSafeAreaView
  style={
    {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      {
        "borderColor": "black",
        "borderWidth": 0,
        "height": 1334,
        "position": "absolute",
        "width": 750,
      }
    }
  />
</RCTSafeAreaView>
`;

exports[`SpinachAppContainer renders without scrollView on IOS 1`] = `
<RCTSafeAreaView
  style={
    {
      "borderColor": "blue",
      "borderWidth": 0,
      "flex": 1,
    }
  }
>
  <Image
    resizeMode="cover"
    source={1}
    style={
      {
        "borderColor": "black",
        "borderWidth": 0,
        "height": 1334,
        "position": "absolute",
        "width": 750,
      }
    }
  />
</RCTSafeAreaView>
`;
